src/charm.py

```diff
@@ -87,8 +95,8 @@ def _configure_sdcore_nms(self, event: EventBase) -> None:
         if not self._sdcore_management.management_url:
             self.unit.status = WaitingStatus("Waiting for webui management url to be available")
             return
-        self._configure_upf_information()
-        self._configure_gnb_information()
+        self._configure_upf_information(event)
```

Comment:
I don't think it's a good idea to pass the event around. We don't really need the event. All we need to do is go through all the existing relations and generate the config file.

Reply:
Right now, with only the relations we do not have all the information we need to identify whether the relation is broken. That's why I am passing the event around.